File: client/src/models/vacancy/vacancy.update.dto.ts
import { AttendanceMode } from "../common/attendance.enum";
import { LocationDto } from "../common/location.dto";
import { SkillDto } from "../common/skill.dto";
import { Experience } from "./experience.enum";

export interface VacancyUpdateModel {
    id: string;
    categoryId: string;
    title: string;
    positionTitle: string;
    description: string;
    salary: number;
    experience: Experience;
    attendanceMode: AttendanceMode;
    locations: LocationDto[];
    skills: SkillDto[];
}

File: src/VacanciesService/Domain/DTO/VacancyUpdateDto.cs
using VacanciesService.Domain.Enums;

namespace VacanciesService.Domain.DTO;

public class VacancyUpdateDto
{
    public Guid Id { get; set; }
    public Guid CategoryId { get; set; }
    public string Title { get; set; }
    public string PositionTitle { get; set; }
    public string Description { get; set; }
    public double Salary { get; set; }
    public Experience Experience { get; set; }
    public AttendanceMode AttendanceMode { get; set; }
    public List<LocationDto> Locations { get; set; }
    public List<SkillDto> Skills { get; set; }
}